### ISSUE TITLE
Improve error message for when dotnet.exe can't be found

### ISSUE THREAD

**Issue by tyilo** [gluck/il-repack#367]
When `dotnet.exe` can't be found, ILRepack fails without explaining that this is the cause:

```
  ------------- IL Repack Arguments -------------
  /out:bin\Release\merged\InstallationProcess.exe /wildcards /target:exe bin\Release\InstallationProcess.exe bin\Releas
  e\*.dll
  -----------------------------------------------
EXEC : warning : Ignoring duplicate resource ILLink.Substitutions.xml [Z:\windows-software-updater\InstallationProcess\
InstallationProcess.csproj]
  System.AggregateException: One or more errors occurred. ---> System.ComponentModel.Win32Exception: The system cannot
  find the file specified
     at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
     at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
     at ILRepacking.ProcessRunner.<RunAsync>d__3.MoveNext() in /_/ILRepack/ProcessRunner.cs:line 32
     --- End of inner exception stack trace ---
     at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
     at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
     at ILRepacking.ProcessRunner.Run(String executableName, String arguments, String workingDirectory) in /_/ILRepack/
  ProcessRunner.cs:line 19
     at ILRepacking.RepackAssemblyResolver.InitializeDotnetRuntimeDirectories() in /_/ILRepack/RepackAssemblyResolver.c
  s:line 187
     at ILRepacking.RepackAssemblyResolver.TryResolveFromCore(AssemblyNameReference reference) in /_/ILRepack/RepackAss
  emblyResolver.cs:line 152
     at ILRepacking.RepackAssemblyResolver.TryResolveFromCoreFixVersion(AssemblyNameReference reference) in /_/ILRepack
  /RepackAssemblyResolver.cs:line 129
     at ILRepacking.RepackAssemblyResolver.TryResolve(AssemblyNameReference name, ReaderParameters parameters) in /_/IL
  Repack/RepackAssemblyResolver.cs:line 112
     at ILRepacking.RepackAssemblyResolver.Resolve(AssemblyNameReference name, ReaderParameters parameters) in /_/ILRep
  ack/RepackAssemblyResolver.cs:line 82
     at Mono.Cecil.MetadataResolver.Resolve(TypeReference type) in /_/Mono.Cecil/MetadataResolver.cs:line 111
     at Mono.Cecil.TypeReference.Resolve() in /_/Mono.Cecil/TypeReference.cs:line 278
     at Mono.Cecil.Mixin.CheckedResolve(TypeReference self) in /_/Mono.Cecil/TypeReference.cs:line 345
     at Mono.Cecil.MetadataBuilder.GetConstantType(TypeReference constant_type, Object constant) in /_/Mono.Cecil/Assem
  blyWriter.cs:line 1899
     at Mono.Cecil.MetadataBuilder.AddConstant(IConstantProvider owner, TypeReference type) in /_/Mono.Cecil/AssemblyWr
  iter.cs:line 1886
     at Mono.Cecil.MetadataBuilder.AddParameter(UInt16 sequence, ParameterDefinition parameter, ParamTable table) in /_
  /Mono.Cecil/AssemblyWriter.cs:line 1775
     at Mono.Cecil.MetadataBuilder.AddParameters(MethodDefinition method) in /_/Mono.Cecil/AssemblyWriter.cs:line 1720
     at Mono.Cecil.MetadataBuilder.AddMethod(MethodDefinition method) in /_/Mono.Cecil/AssemblyWriter.cs:line 1688
     at Mono.Cecil.MetadataBuilder.AddMethods(TypeDefinition type) in /_/Mono.Cecil/AssemblyWriter.cs:line 1672
     at Mono.Cecil.MetadataBuilder.AddType(TypeDefinition type) in /_/Mono.Cecil/AssemblyWriter.cs:line 1473
     at Mono.Cecil.MetadataBuilder.AddTypes() in /_/Mono.Cecil/AssemblyWriter.cs:line 1444
     at Mono.Cecil.MetadataBuilder.BuildTypes() in /_/Mono.Cecil/AssemblyWriter.cs:line 1295
     at Mono.Cecil.MetadataBuilder.BuildModule() in /_/Mono.Cecil/AssemblyWriter.cs:line 1066
     at Mono.Cecil.MetadataBuilder.BuildMetadata() in /_/Mono.Cecil/AssemblyWriter.cs:line 1036
     at Mono.Cecil.ModuleWriter.Write(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters) in /_/
  Mono.Cecil/AssemblyWriter.cs:line 121
     at Mono.Cecil.ModuleWriter.WriteModule(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters)
  in /_/Mono.Cecil/AssemblyWriter.cs:line 78
     at Mono.Cecil.ModuleDefinition.Write(String fileName, WriterParameters parameters) in /_/Mono.Cecil/ModuleDefiniti
  on.cs:line 1173
     at ILRepacking.ILRepack.RepackCore(String tempOutputDirectory) in /_/ILRepack/ILRepack.cs:line 432
     at ILRepacking.ILRepack.Repack() in /_/ILRepack/ILRepack.cs:line 271
     at ILRepacking.Application.Main(String[] args) in /_/ILRepack/Application.cs:line 28
  ---> (Inner Exception #0) System.ComponentModel.Win32Exception (0x80004005): The system cannot find the file specifie
  d
     at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
     at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
     at ILRepacking.ProcessRunner.<RunAsync>d__3.MoveNext() in /_/ILRepack/ProcessRunner.cs:line 32<---
```

I spent a lot of time figuring out why ILRepack wasn't working on one of our machines. Having a log statement saying "Couldn't find dotnet.exe" would have been very useful.

**Comment by KirillOsenkov** [comment]
Hopefully this is better:
![image](https://github.com/user-attachments/assets/43276946-47aa-4e41-92a0-d4918d354171)

**Comment by KirillOsenkov** [comment]
Fixed in https://github.com/gluck/il-repack/commit/ccc51425b903018e377bf63e05d10e75124bbe87

**Comment by tyilo** [comment]
> Fixed in [ccc5142](https://github.com/gluck/il-repack/commit/ccc51425b903018e377bf63e05d10e75124bbe87)

Thanks! ❤️
